remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RuleSetsImpl.java

```diff
@@ -110,6 +111,22 @@ public RuleSetLink linkToRuleSet(String folderNodeId, String linkToNodeId)
         return ruleSetLink;
     }
 
+    @Override
+    public void unlinkRuleSet(String folderNodeId, String ruleSetId)
+    {
+        final NodeRef folderNodeRef = validator.validateFolderNode(folderNodeId,true);
+        final NodeRef ruleSetNodeRef = validator.validateRuleSetNode(ruleSetId, folderNodeRef);
+
+        //The folder should be linked to a rule set
+        if (!ruleService.isLinkedToRuleNode(folderNodeRef))
+        {
+            throw new InvalidArgumentException("The folder is not linked to a rule set.");

```

Comment:
This is interesting. I was expecting us to just return successfully if the link doesn't exist.  Would it be possible to check whether other DELETE endpoints can throw a 404?

Reply:
Looks like we're also throwing a 404 in the endpoint for deleting rules, when the rule doesn't exist.

Reply:
Ok - that's brilliant :+1: